libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/ActionScope.cs

```diff
@@ -202,7 +202,6 @@ protected virtual async Task<DialogTurnResult> BeginActionAsync(DialogContext dc
             {
                 { "DialogId", action.Id },
                 { "Kind", $"Microsoft.{actionName}" },
-                { "Instance", JsonConvert.SerializeObject(action, new JsonSerializerSettings() { NullValueHandling = NullValueHandling.Ignore }).ToString() }
```

Comment:
@garypretty What do you suggest here that is not serializing the entire action? It has a reference to the dialog that could be circular or just large.

Reply:
For other folks - while trying this code with rather large json structures we found that this line was extremely expensive and potentially dangerous as it could lead to large object heap usage and it's simply a lot of potentially unnecessary serialization.

Reply:
I would like @vishwacsena to comment on this.  @vishwacsena has asked for all of the properties / scope for the action to be captured as part of telemetry.  I am not familiar enough with adaptive right now to definitively say what we should / should not capture here.

Reply:
Again, this is actually serializing ALL of the actions, recursively doing a deep clone of all child settings, recognizers, etc.  Not good at all

---
In reply to: [419414372](https://github.com/microsoft/botbuilder-dotnet/pull/3857#discussion_r419414372) [](ancestors = 419414372)

Reply:
I think for this PR, removing the Instance property is enough.  We can always make additive changes to the captured telemetry in a future release. I think we're good.